tests/subsys/bluetooth/controller/rng/src/main.c

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <ztest.h>
+#include <stdint.h>
+#include <soc.h>
+#include <logging/log.h>
+#include "nrf_errno.h"
+#include "ble_controller_soc.h"
+#include <entropy.h>
+#include <mock_ext.h>
+#include <kernel.h>
+
+#define LOG_MODULE_NAME test_ble_controller_rng
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, LOG_LEVEL_DBG);
+
+#define SEM_GIVE_ERROR (-0xFF)
+#define RANDOM_POOL_CAPACITY (100)
+
+/* The ble controller poll length is limited to UINT8_MAX. This should be
+ * more than uint8_t to ensure it fills the whole buffer. */
+#define BUFFER_LENGTH 1024
+static u8_t buffer[BUFFER_LENGTH];
+static uint8_t rand_pool[BUFFER_LENGTH];
+
+/* This could only be a macro because ztest_returns_value() uses __func__ inside
+ * to get the stored value. */
+#define rand_pool_to_buffer_copy(p_dst, length)								 \
```

Comment:
Please use a function instead of a macro

Reply:
It is not possible. The comment is above the macro :)